packages/client/src/components/SendTokens/steps/DepositTokenForm.js

```diff
@@ -0,0 +1,101 @@
+import { useContext, useEffect, useState } from 'react';
+import { Web3Context } from 'contexts/Web3';
+import AmountInput from 'components/SendTokens/components/AmountInput';
+import AssetSelector from 'components/SendTokens/components/AssetSelector';
+import { SendTokensContext } from 'components/SendTokens/sendTokensContext';
+import { useAccount, useContacts } from 'hooks';
+import { ASSET_TYPES } from 'constants/enums';
+import { formatAddress, getNameByAddress, isVaultInTreasury } from 'utils';
+import Svg from 'library/Svg';
+
+const DepositTokenForm = ({ safeData, userAddress }) => {
+  const [sendModalState] = useContext(SendTokensContext);
+  const { assetType } = sendModalState;
+  const { address: safeAddress, name: safeName } = safeData;
+  const { contacts } = useContacts(safeAddress);
+  const web3 = useContext(Web3Context);
+  const { getUserBalances } = useAccount();
+  const [coinBalances, setCoinBalances] = useState();
+
+  const { getTreasuryVaults, vaults } = web3;
+
+  const getVaults = async () => {
+    await getTreasuryVaults(safeAddress);
+  };
+
+  const updateUserBalance = async () => {
+    try {
+      const userBalances = await getUserBalances(formatAddress(userAddress));
+      const balances = {};
+      userBalances.forEach((userBalance) => {
+        const { coinType, balance } = userBalance;
+        if (isVaultInTreasury(vaults[safeAddress], coinType)) {
+          balances[userBalance.coinType] = balance;
+        }
+      });
+
+      setCoinBalances(balances);
+    } catch (err) {
+      console.log(`Failed to get coin balances, error: ${err}`);
+    }
+  };
+
+  useEffect(() => {
+    if (!userAddress) {
+      return;
+    }
+    getVaults();
+    // eslint-disable-next-line
+  }, [userAddress]);
+
+  useEffect(() => {
```

Comment:
> > > @KsenijaZivkovic re `Send button on Tokens page is hidden if the vault is empty` , wasn't aware that we need to hide 'send' button when vault is empty, could you help clarify if that's a new design change request from Alaa?
> > 
> > 
> > @amyliu6 I have proposed this change, because I feel that it shouldn't be possible to send tokens from the empty vault. Even if the user opens the vault they wouldn't be able to change the amount and BE also prevents the users to propose 0 tokens to account. Let me know your thoughts, I can easily revert it!
> 
> That true. As there were design exploration to notify user about empty vault in "Send flow" (for future release), hence want to double check here, we should only update design after getting their approval.

Updated as per design, to disable buttons instead!

Reply:
> Can you help me understand the purpose of this effect? Looks like it will only be called on the deposit token modal open since vaults won't change during the depositing flow?

It was the only way to sync treasury vaults and user vaults -> to fetch the vaults from the treasury first and then to update the vaults of the user that can be used to deposit tokens.

Reply:
I guess I'm missing the point of having two separate useEffect hooks, why not combine with the hook above?

Reply:
I'll try again and change it if it works in one useEffect hook.

Reply:
It causes infinite loops when merged into one useEffect. The use case that this is solving is when user opens the page for the first time and local storage is empty, when vaults are loaded to update the user balance

Reply:
can we do something like this?
```
useEffect(()=>{
    if(isEmpty(vaults)){
         getTreasuryVaults();
    }
    updateUserBalance();
}, [userAddress])
```

Reply:
Updated it like this:
```
  useEffect(() => {
    if (!userAddress) {
      return;
    }
    if (isEmpty(vaults)) {
      getVaults();
    } else {
      updateUserBalance();
    }
  }, [userAddress, vaults]);
  ```